homeassistant/components/onvif/camera.py

```diff
@@ -194,33 +194,36 @@ def __init__(self, hass, config):
             system_date = dt_util.utcnow()
             device_time = await devicemgmt.GetSystemDateAndTime()
             if device_time:
```

Comment:
Please rewrite this code to guard clause. 

```python
if not device_time:
    return

if cdate is None:
    return
```

Reply:
Thanks @ballod, I will put the change so that the time is set to `UTC` or `Local`. I will open another pull, for implementation of a parameter that defines that a certain camera is required to use `UTC` conversion, because for example, in this mine (Chinese), it only comes with the field `LocalDateTime` and it is in `UTC` time. , not an exclusive for my camera I believe, as there may be other companies that are not officially Onvif partners, performing incomplete implementations of the protocol.

Reply:
In that case we should keep a list of models that need this quirk applied.  We can't assume incorrect application for all

Reply:
The problem is that we have no way to check the camera being analyzed to mount a filter. I was thinking of an optional parameter in `configuration.yaml` that the user could set the camera time to be` UTC` instead of using TimeZone set up for Home Assistant. What do you think? I can open an issue to discuss this if you find it interesting.

Reply:
Can't we query the manufacturer and model of the camera?

Reply:
I will read the documentation about and see what I find, but I did a test once, and there are few functions implemented by these types of companies, so I think a filter for them will not fit. But ... if anyone reading has any suggestions, you can comment.

Reply:
@balloob I was able to collect the information, but it does not seem very safe for a filter. Unfortunately I have no other cameras to check if a homologated information would be more accurate. But what do you think, would you implement by verifying this information?

```
2019-10-20 14:11:35 DEBUG (MainThread) [homeassistant.components.onvif.camera] {
    'Manufacturer': 'IPCAM',
    'Model': 'IPCAM',
    'FirmwareVersion': 'HS-Camera_No1',
    'SerialNumber': '74******8c',
    'HardwareId': '00059a********************a14018c'
}
```